tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,321 @@
+		Region:                    backupRestoreConfig.S3Region,
+	}
+
+	_, err = createOpaqueS3Secret(client.Steve)
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("Installing backup chart")
+	err = charts.InstallRancherBackupChart(client, chartInstallOptions, chartFeatureOptions, true)
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("Waiting for backup chart deployments to have expected number of available replicas")
+	err = charts.WatchAndWaitDeployments(client, project.ClusterID, backupChartNamespace, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func createOpaqueS3Secret(steveClient *v1.Client) (string, error) {
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	logrus.Infof("Creating an opaque secret with name: %v", secretName)
+	secretTemplate := secrets.NewSecretTemplate(secretName, backupRestoreConfig.CredentialSecretNamespace, map[string][]byte{"accessKey": []byte(backupRestoreConfig.AccessKey), "secretKey": []byte(backupRestoreConfig.SecretKey)}, corev1.SecretTypeOpaque)
+	createdSecret, err := steveClient.SteveType(secrets.SecretSteveType).Create(secretTemplate)
+
+	return createdSecret.Name, err
+}
+
+func createProject(client *rancher.Client, clusterID string) (*management.Project, error) {
+	createProject, _, err := projectsapi.CreateProjectAndNamespace(client, clusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	return createProject, nil
+}
+
+func createRole(client *rancher.Client, context string, roleName string, rules []management.PolicyRule) (role *management.RoleTemplate, err error) {
+	role, err = client.Management.RoleTemplate.Create(
+		&management.RoleTemplate{
+			Context: context,
+			Name:    roleName,
+			Rules:   rules,
+		})
+
+	return
+}
+
+func createRancherResources(client *rancher.Client, resourceCount int, clusterID string) ([]*management.User, []*management.Project, []*management.RoleTemplate, error) {
+	userList := []*management.User{}
+	projList := []*management.Project{}
+	roleList := []*management.RoleTemplate{}
+
+	for i := 0; i < resourceCount; i++ {
+		u, err := users.CreateUserWithRole(client, users.UserConfig(), globalUserRole)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		userList = append(userList, u)
+
+		p, err := createProject(client, clusterID)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		projList = append(projList, p)
+
+		rt, err := createRole(client, clusterRoleContext, roleName, rules)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		roleList = append(roleList, rt)
+	}
+
+	return userList, projList, roleList, nil
+}
+
+func createBackup(client *rancher.Client, name string) (*v1.SteveAPIObject, error) {
+	backup := setBackupObject()
+	backupTemplate := bv1.NewBackup("", name, backup)
+	completedBackup, err := client.Steve.SteveType(backupSteveType).Create(backupTemplate)
+
+	return completedBackup, err
```

Comment:
supernit; this might already be the right format but I'm curious, could you just return the function call to steve.create instead of declaring vars first? 

the var name `completedBackup` isn't quite accurate since this is just asking steve to start the backup.